### ISSUE TITLE
7-hour long program not downloaded completely

### ISSUE THREAD

**Issue by uusijani** [aajanki/yle-dl#290]
Trying to download [this 7-hour long program](https://areena.yle.fi/1-50934704) only downloads (about) the first 6 hours and 40 minutes (~ 11 GB), after which the process just stops, and leaves Gnome terminal screwed up (typed characters are invisible until I run `reset`). In the web player the video seems to play past that point just fine.

This is on Ubuntu 20.04 with ffmpeg version N-58594-g715f63232f-static (static build from git master on 20210908).

    jani@saegusa:Työpöytä$ /usr/local/bin/yle-dl 'https://areena.yle.fi/1-50934704'
    yle-dl 20210808: Download media files from Yle Areena and Elävä Arkisto
    Copyright (C) 2009-2021 Antti Ajanki <antti.ajanki@iki.fi>, license: GPLv3

    Unsupported codec with id 98313 for input stream 2
    Unsupported codec with id 98313 for input stream 5
    Unsupported codec with id 98313 for input stream 8
    Unsupported codec with id 98313 for input stream 11
    Unsupported codec with id 98313 for input stream 14
    Unsupported codec with id 98313 for input stream 17
    Unsupported codec with id 98313 for input stream 20
    Unsupported codec with id 98313 for input stream 23
    Unsupported codec with id 98313 for input stream 26
    Unsupported codec with id 98313 for input stream 29
    Output file: YleX Esittää: Yle 95 - synttäribileet Linkkitornista: 2021-09-18T00:00.mkv
    jani@saegusa:Työpöytä$  size=10814976kB time=06:39:40.68 bitrate=3694.5kbits/s speed=  44x

Note the prompt appearing on top of ffmpeg output on the last line. Typing `echo $?` reveals that the process exited with code 1.

Here's [the output when run with `--verbose`](https://github.com/aajanki/yle-dl/files/7197916/ylex-verbose.log).

I've only tested this over the past couple of days so I don't know if it's a temporary error. Since the portion of the stream that I actually care about is all prior to the 6-hour mark, and the partial download is playable, I'm not too bothered to have the rest download. Reporting this just in case it's a useful test of a corner case.

**Comment by aajanki** [comment]
There has been similar reports on very longprograms also before. I don't know what is causing this. A work-around (obviously not ideal) is to download a low resolution variant, for example `--resolution 540`.

**Comment by uusijani** [comment]
All right, I got curious enough try again, and found that this is apparently caused by the [static build of ffmpeg](https://johnvansickle.com/ffmpeg/) that I'm using: I used `yle-dl --verbose` again, then interrupted the download immediately, ran the ffmpeg command (as reported by yle-dl) directly, and this resulted in a segmentation fault.

I then tried the same command in a VM with ffmpeg from Ubuntu repositories, and there the download finished successfully.

So the only remaining issue with yle-dl in this is the terminal being broken: that did not occur with directly-run ffmpeg despite the segfault.

I'm closing this issue though, so that it remains as documentation for the problem (and cause) as originally reported.
